### ISSUE TITLE
RuntimeError: Expected a 'cuda' device type for generator but found 'cpu'

### ISSUE THREAD

**Issue by ashrayar** [ml-struct-bio/cryodrgn#117]
I installed cryodrgn as per the instructions given. According to the nvidia-smi output, the Cuda version is 11.5 but the latest cudatoolkit version I could install along with pytorch was 11.3. I checked out cryodrgn version 0.3.4. When I ran the quicktest.sh script, I get the following output:
```
+ cryodrgn train_vae data/hand.mrcs -o output/toy_recon_vae --lr .0001 --seed 0 --poses data/hand_rot.pkl --zdim 10
2022-06-09 18:55:51     /home/aravikumar/anaconda3/envs/cryodrgn/bin/cryodrgn train_vae data/hand.mrcs -o output/toy_recon_vae --lr .0001 --seed 0 --poses data/hand_rot.pkl --zdim 10
2022-06-09 18:55:51     Namespace(activation='relu', amp=False, batch_size=8, beta=None, beta_control=None, checkpoint=1, ctf=None, datadir=None, do_pose_sgd=False, domain='fourier', emb_type='quat', enc_mask=None, encode_mode='resid', func=<function main at 0x7f071a69e9e0>, ind=None, invert_data=True, lazy=False, load=None, log_interval=1000, lr=0.0001, max_threads=16, multigpu=False, norm=None, num_epochs=20, outdir='/home/aravikumar/software/cryodrgn/testing/output/toy_recon_vae', particles='/home/aravikumar/software/cryodrgn/testing/data/hand.mrcs', pdim=256, pe_dim=None, pe_type='geom_lowf', players=3, pose_lr=0.0003, poses='/home/aravikumar/software/cryodrgn/testing/data/hand_rot.pkl', preprocessed=False, pretrain=1, qdim=256, qlayers=3, relion31=False, seed=0, tilt=None, tilt_deg=45, use_real=False, verbose=False, wd=0, window=True, window_r=0.85, zdim=10)
2022-06-09 18:55:51     Use cuda True
2022-06-09 18:55:51     Loading dataset from /home/aravikumar/software/cryodrgn/testing/data/hand.mrcs
2022-06-09 18:55:51     Loaded 100 64x64 images
2022-06-09 18:55:51     Windowing images with radius 0.85
2022-06-09 18:55:51     Computing FFT
2022-06-09 18:55:51     Spawning 16 processes
2022-06-09 18:55:51     Symmetrizing image data
2022-06-09 18:55:51     Normalized HT by 0 +/- 94.426513671875
2022-06-09 18:55:51     WARNING: No translations provided
2022-06-09 18:55:59     Using circular lattice with radius 32
2022-06-09 18:55:59     HetOnlyVAE(
  (encoder): ResidLinearMLP(
    (main): Sequential(
      (0): Linear(in_features=3208, out_features=256, bias=True)
      (1): ReLU()
      (2): ResidLinear(
        (linear): Linear(in_features=256, out_features=256, bias=True)
      )
      (3): ReLU()
      (4): ResidLinear(
        (linear): Linear(in_features=256, out_features=256, bias=True)
      )
      (5): ReLU()
      (6): ResidLinear(
        (linear): Linear(in_features=256, out_features=256, bias=True)
      )
      (7): ReLU()
      (8): Linear(in_features=256, out_features=20, bias=True)
    )
  )
  (decoder): FTPositionalDecoder(
    (decoder): ResidLinearMLP(
      (main): Sequential(
        (0): Linear(in_features=202, out_features=256, bias=True)
        (1): ReLU()
        (2): ResidLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (3): ReLU()
        (4): ResidLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (5): ReLU()
        (6): ResidLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (7): ReLU()
        (8): Linear(in_features=256, out_features=2, bias=True)
      )
    )
  )
)
2022-06-09 18:55:59     1273878 parameters in model
2022-06-09 18:55:59     1024020 parameters in encoder
2022-06-09 18:55:59     249858 parameters in decoder
Traceback (most recent call last):
  File "/home/aravikumar/anaconda3/envs/cryodrgn/bin/cryodrgn", line 33, in <module>
    sys.exit(load_entry_point('cryodrgn==0.3.4', 'console_scripts', 'cryodrgn')())
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/cryodrgn-0.3.4-py3.7.egg/cryodrgn/__main__.py", line 54, in main
    args.func(args)
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/cryodrgn-0.3.4-py3.7.egg/cryodrgn/commands/train_vae.py", line 424, in main
    for minibatch in data_generator:
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 530, in __next__
    data = self._next_data()
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 569, in _next_data
    index = self._next_index()  # may raise StopIteration
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 521, in _next_index
    return next(self._sampler_iter)  # may raise StopIteration
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/torch/utils/data/sampler.py", line 226, in __iter__
    for idx in self.sampler:
  File "/home/aravikumar/anaconda3/envs/cryodrgn/lib/python3.7/site-packages/torch/utils/data/sampler.py", line 122, in __iter__
    yield from torch.randperm(n, generator=generator).tolist()
RuntimeError: Expected a 'cuda' device type for generator but found 'cpu'
```

**Comment by ashrayar** [comment]
I did that and now get the following error:

```
+ cryodrgn train_vae data/hand.mrcs -o output/toy_recon_vae --lr .0001 --seed 0 --poses data/hand_rot.pkl --zdim 10 --pe-type gaussian
usage: cryodrgn train_vae [-h] -o OUTDIR --zdim ZDIM --poses POSES [--ctf pkl]
                          [--load WEIGHTS.PKL] [--checkpoint CHECKPOINT]
                          [--log-interval LOG_INTERVAL] [-v] [--seed SEED]
                          [--ind PKL] [--uninvert-data] [--no-window]
                          [--window-r WINDOW_R] [--datadir DATADIR]
                          [--relion31] [--lazy] [--preprocessed]
                          [--max-threads MAX_THREADS] [--tilt TILT]
                          [--tilt-deg TILT_DEG] [-n NUM_EPOCHS]
                          [-b BATCH_SIZE] [--wd WD] [--lr LR] [--beta BETA]
                          [--beta-control BETA_CONTROL] [--norm NORM NORM]
                          [--amp] [--multigpu] [--do-pose-sgd]
                          [--pretrain PRETRAIN] [--emb-type {s2s2,quat}]
                          [--pose-lr POSE_LR] [--enc-layers QLAYERS]
                          [--enc-dim QDIM]
                          [--encode-mode {conv,resid,mlp,tilt}]
                          [--enc-mask ENC_MASK] [--use-real]
                          [--dec-layers PLAYERS] [--dec-dim PDIM]
                          [--pe-type {geom_ft,geom_full,geom_lowf,geom_nohighf,linear_lowf,none}]
                          [--pe-dim PE_DIM] [--domain {hartley,fourier}]
                          [--activation {relu,leaky_relu}]
                          particles
cryodrgn train_vae: error: argument --pe-type: invalid choice: 'gaussian' (choose from 'geom_ft', 'geom_full', 'geom_lowf', 'geom_nohighf', 'linear_lowf', 'none')
(cryodrgn) bash-4.4$ cryodrgn -h
usage: cryodrgn [-h] [--version]
                {downsample,preprocess,parse_pose_csparc,parse_pose_star,parse_ctf_csparc,parse_ctf_star,train_nn,backproject_voxel,train_vae,eval_vol,eval_images,analyze,pc_traversal,graph_traversal,view_config}
                ...

CryoDRGN neural network reconstruction

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

Choose a command:
  {downsample,preprocess,parse_pose_csparc,parse_pose_star,parse_ctf_csparc,parse_ctf_star,train_nn,backproject_voxel,train_vae,eval_vol,eval_images,analyze,pc_traversal,graph_traversal,view_config}
```

**Comment by zhonge** [comment]
Apologies I missed your reply. If you haven't figured out the problem, can you check the version with `cryodrgn --version`? The new positional encoding type `--pe-type gaussian` should be available in version 1.0.0-beta.

**Comment by ashrayar** [comment]
Hi, it worked now. I guess I didn't run the setup.py script after checking out the beta branch. Sorry for the trouble! Thanks
